cmd/up/exec.go

```diff
@@ -262,6 +272,26 @@ func getDefaultLocalEnvs() []string {
 	return envs
 }
 
+func (d *devContainerEnvGetter) getEnvsFromDevContainer(ctx context.Context, spec *apiv1.PodSpec, name, namespace string, client kubernetes.Interface) ([]string, error) {
+	var envs []string
+
+	devContainer := apps.GetDevContainer(spec, name)
+
+	for _, env := range devContainer.Env {
+		val := env.Value
+		if env.ValueFrom != nil && env.ValueFrom.SecretKeyRef != nil {
```

Comment:
NIT: I don't think it should be covered in this PR but wanted to mention that there is a common scenario where a user adds env vars from a configmap by using `configMapKeyRef`, just like the secrets. But let's keep the scope short

Reply:
@jLopezbarb good catch, I've created a follow-up issue to keep scope down: https://github.com/okteto/okteto/issues/3936

Reply:
I've opened a new PR for the configmap change: https://github.com/okteto/okteto/pull/3937

Reply:
Nice!!